pkg/controller/acmechallenges/controller.go

```diff
@@ -82,6 +76,10 @@ type controller struct {
 	dns01Nameservers []string
 
 	DNS01CheckRetryPeriod time.Duration
+
+	// objectUpdater implements the updateObject function which is used to save
+	// changes to the Challenge.Status and Challenge.Finalizers
+	objectUpdater
```

Comment:
This is nice, I think we could use the same approach in constructing all controller eventually